apps/main/src/llamalend/features/user-position-history/hooks/useUserLendCollateralEvents.ts

```diff
@@ -0,0 +1,114 @@
+import { useMemo } from 'react'
+import { useUserLendCollateralEventsQuery } from '@/llamalend/features/user-position-history/queries/user-lend-collateral-events'
+import { IChainId } from '@curvefi/llamalend-api/lib/interfaces'
+import { UserCollateralEvent, UserCollateralEvents } from '@curvefi/prices-api/lending'
+
+/**
+ * Types:
+ * Add collateral = "Borrow" type when collateral incresases but debt doesn't.
+ * Self liquidation = "Liquidate" when the liquidator is the user.
+ * Hard liquidation = "Liquidate" when the liquidator is not the user.
+ * Borrow more = "Borrow" type when debt increases but collateral doesn't.
+ * Repay and Close = "Repay" when debt goes to 0.
+ */
+export type UserLendCollateralEventType =
+  | 'Open Position'
+  | 'Borrow'
+  | 'Liquidate'
+  | 'Repay'
+  | 'Repay and Close'
+  | 'Remove Collateral'
+  | 'Borrow More'
+  | 'Add Collateral'
+  | 'Self Liquidation'
+  | 'Hard Liquidation'
+
+export type ParsedUserLendCollateralEvent = Omit<UserCollateralEvent, 'type'> & {
+  type: UserLendCollateralEventType
+  url: string
+  borrowToken: {
+    symbol: string
+    address: string
```

Comment:
we should use `Address` as much as possible. But do we really need to declare this type again I think we have several types for this :dizzy_face: 
```suggestion
    address: Address
```